### PR TITLE
mod_event_pusher_rabbit quickfix

### PR DESCRIPTION
**Pull request by NelsonVides**
Simple quickfix, seems to be receiving something related to the users entering the room.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[7175.1](https://travis-ci.org/esl/MongooseIM/jobs/584679168) / Erlang 22.0 / small_tests / a1b50e8d7aa68c4e73655b83af27e8e3e66fe708
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2455/7175/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2455/7175/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-09-13_18.16.04/lib.mongooseim.logs/run.2019-09-13_18.16.04/suite.log.html)

File: big_tests/tests/mod_event_pusher_rabbit_SUITE.erl [hunk 1]
```diff
@@ -414,20 +414,20 @@ group_chat_message_received_event_properly_formatted(Config) ->
               BobRoomJID = user_room_jid(RoomAddr, Bob),
               AliceJID = client_lower_short_jid(Alice),
               AliceFullJID = client_lower_full_jid(Alice),
-              AliceGroupChatMsgRecvRK = group_chat_msg_recv_rk(AliceJID),
-              Message = <<"Hi there!">>,
-              listen_to_group_chat_msg_recv_events_from_rabbit([AliceJID],
-                                                               Config),
               %% WHEN users chat
```
Comment (mkacper) [inline big_tests/tests/mod_event_pusher_rabbit_SUITE.erl]: I'd update the comment - here we can add some explanation why users first join the room and only then we subscribe to Rabbit.

File: big_tests/tests/mod_event_pusher_rabbit_SUITE.erl [hunk 2]
```diff
@@ -414,20 +414,20 @@ group_chat_message_received_event_properly_formatted(Config) ->
...
+              AliceGroupChatMsgRecvRK = group_chat_msg_recv_rk(AliceJID),
+              Message = <<"Hi there!">>,
+              listen_to_group_chat_msg_recv_events_from_rabbit([AliceJID],
+                                                               Config),
 
               escalus:send(Bob, escalus_stanza:groupchat_to(RoomAddr, Message)),
               %% THEN
               %% TODO: Investigate why there is an empty message sent.
-              get_decoded_message_from_rabbit(AliceGroupChatMsgRecvRK),
+              % get_decoded_message_from_rabbit(AliceGroupChatMsgRecvRK),
```
Comment (mkacper) [inline big_tests/tests/mod_event_pusher_rabbit_SUITE.erl]: I'd just remove that function call.

File: big_tests/tests/mod_event_pusher_rabbit_SUITE.erl [hunk 3]
```diff
@@ -414,20 +414,20 @@ group_chat_message_received_event_properly_formatted(Config) ->
...
               escalus:send(Bob, muc_helper:stanza_muc_enter_room(Room,
                                                                  nick(Bob))),
+              AliceGroupChatMsgRecvRK = group_chat_msg_recv_rk(AliceJID),
+              Message = <<"Hi there!">>,
+              listen_to_group_chat_msg_recv_events_from_rabbit([AliceJID],
+                                                               Config),
 
               escalus:send(Bob, escalus_stanza:groupchat_to(RoomAddr, Message)),
               %% THEN
               %% TODO: Investigate why there is an empty message sent.
```
Comment (mkacper) [inline big_tests/tests/mod_event_pusher_rabbit_SUITE.erl]: I think that this comment is no longer relevant after your excellent investigation 🥇

File: big_tests/tests/mod_event_pusher_rabbit_SUITE.erl [hunk 4]
```diff
@@ -414,20 +414,20 @@ group_chat_message_received_event_properly_formatted(Config) ->
...
-                                                               Config),
               %% WHEN users chat
               escalus:send(Alice, muc_helper:stanza_muc_enter_room(Room,
                                                                    nick(Alice))),
               escalus:send(Bob, muc_helper:stanza_muc_enter_room(Room,
                                                                  nick(Bob))),
+              AliceGroupChatMsgRecvRK = group_chat_msg_recv_rk(AliceJID),
+              Message = <<"Hi there!">>,
+              listen_to_group_chat_msg_recv_events_from_rabbit([AliceJID],
+                                                               Config),
```
Comment (mkacper) [inline big_tests/tests/mod_event_pusher_rabbit_SUITE.erl]: We can move the comment `%% WHEN users chat` here :)

File: big_tests/tests/mod_event_pusher_rabbit_SUITE.erl [hunk 5]
```diff
@@ -414,20 +414,20 @@ group_chat_message_received_event_properly_formatted(Config) ->
...
-              AliceGroupChatMsgRecvRK = group_chat_msg_recv_rk(AliceJID),
-              Message = <<"Hi there!">>,
-              listen_to_group_chat_msg_recv_events_from_rabbit([AliceJID],
-                                                               Config),
               %% WHEN users chat
               escalus:send(Alice, muc_helper:stanza_muc_enter_room(Room,
                                                                    nick(Alice))),
               escalus:send(Bob, muc_helper:stanza_muc_enter_room(Room,
                                                                  nick(Bob))),
+              AliceGroupChatMsgRecvRK = group_chat_msg_recv_rk(AliceJID),
```
Comment (mkacper) [inline big_tests/tests/mod_event_pusher_rabbit_SUITE.erl]: Actually I'd leave this and the next line in `GIVEN` section.

**Comment by mongoose-im** [comment]
[7178.1](https://travis-ci.org/esl/MongooseIM/jobs/585459235) / Erlang 22.0 / small_tests / 40a5e3cd19542b3dde6f28670876cffe66c2652f
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2455/7178/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2455/7178/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-09-16_08.11.30/lib.mongooseim.logs/run.2019-09-16_08.11.31/suite.log.html)

File: big_tests/tests/mod_event_pusher_rabbit_SUITE.erl [hunk 6]
```diff
@@ -419,6 +419,9 @@ group_chat_message_received_event_properly_formatted(Config) ->
               %% GIVEN users in room
               escalus:send(Alice, muc_helper:stanza_muc_enter_room(Room, nick(Alice))),
               escalus:send(Bob, muc_helper:stanza_muc_enter_room(Room, nick(Bob))),
+              % wait for all room stanzas to be processed
+              escalus:wait_for_stanzas(Alice, 3),
```
Comment (mkacper) [inline big_tests/tests/mod_event_pusher_rabbit_SUITE.erl]: Why are we waiting for 3 stanzas?

Comment (NelsonVides) [reply big_tests/tests/mod_event_pusher_rabbit_SUITE.erl]: That's how many stanzas the room sends as an answer:
```
alicE94.191079@localhost/res1 out
<presence to='muc_publish@muc.localhost/alicE94.191079'>
	<x xmlns='http://jabber.org/protocol/muc'/>
</presence>

alicE94.191079@localhost/res1 in
<presence from='muc_publish@muc.localhost/alicE94.191079' to='alicE94.191079@localhost/res1' xml:lang='en'>
	<x xmlns='http://jabber.org/protocol/muc#user'>
		<item jid='alicE94.191079@localhost/res1' affiliation='owner' role='moderator'/>
		<status code='110'/>
		<status code='100'/>
		<status code='201'/>
	</x>
</presence>

alicE94.191079@localhost/res1 in
<message from='muc_publish@muc.localhost' to='alicE94.191079@localhost/res1' type='groupchat'>
	<subject/>
	<body/>
</message>

alicE94.191079@localhost/res1 in
<presence from='muc_publish@muc.localhost/bOb94.191079' to='alicE94.191079@localhost/res1' xml:lang='en'>
	<x xmlns='http://jabber.org/protocol/muc#user'>
		<item jid='bOb94.191079@localhost/res1' affiliation='none' role='participant'/>
	</x>
</presence>
```

Comment (NelsonVides) [reply big_tests/tests/mod_event_pusher_rabbit_SUITE.erl]: For example: https://github.com/esl/MongooseIM/blob/6715faa6c82e4965d12e0db3940a0d28b19a4c27/big_tests/tests/muc_SUITE.erl#L685-L690

Comment (mkacper) [reply big_tests/tests/mod_event_pusher_rabbit_SUITE.erl]: Thanks :)

**Comment by mongoose-im** [comment]
[7179.1](https://travis-ci.org/esl/MongooseIM/jobs/585494338) / Erlang 22.0 / small_tests / dfb4a8ccb7f7906ecad5777b9a0bfd62219f5e78
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2455/7179/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2455/7179/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-09-16_09.52.10/lib.mongooseim.logs/run.2019-09-16_09.52.10/suite.log.html)

**Pull request merged**